framework/include/mesh/MooseMesh.h

```diff
@@ -1575,6 +1592,15 @@ class MooseMesh : public MooseObject, public Restartable, public PerfGraphInterf
                             int child,
                             int child_side);
 
+  void buildHRefinementAndCoarseningMaps(Assembly * assembly);
+
+  const std::vector<QpMap> & getPRefinementMapHelper(
```

Comment:
I dont understand yet why it s the mesh's responsibility to handle p-refinement. Seems like a FE thing to me

Reply:
`p-refinement` is a flag on the `Elem`. It is treated analogously to h-refinement